gobblin-utility/src/main/java/org/apache/gobblin/util/filesystem/ExceptionCatchingPathAlterationListenerDecorator.java

```diff
@@ -111,4 +111,13 @@ public void onFileDelete(Path path) {
       log.error("onFileDelete failure: ", exc);
     }
   }
+
+  @Override
+  public void onCheckDetectedChange() {
+    try {
+      this.underlying.onCheckDetectedChange();
+    } catch (Throwable exc) {
+      log.error("onFileDelete failure: ", exc);
+    }
```

Comment:
tip: as the `on*()` methods don't throw checked exceptions, you could more succinctly implement the wrapping using a `Callable` if you define:
```
protected void logSwallowedThrowable(Callable c, String methodName) { ... }
```
note, I'm passing the name explicitly, but you could also use `Throwable.getStackTrace()` to pull out the `StackTraceElement.getMethodName()`

then call as:
```
public void onCheckDetectedChange() {
  logSwallowedThrowable(() -> this.underlying.onCheckDetectedChange(), "onCheckDetectedChange");
}
```
also, note the copy-paste slip of `"onFileDelete"` above

Reply:
Oops good catch, will try to implement this